rust_src/src/windows.rs

```diff
@@ -2034,3 +2035,15 @@ pub fn window_pixel_height_before_size_change(window: LispWindowValidOrSelected)
 }
 
 include!(concat!(env!("OUT_DIR"), "/windows_exports.rs"));
+
+/// Run `window-configuration-change-hook' for FRAME.
+/// If FRAME is omitted or nil, it defaults to the selected frame.
+#[lisp_fn(min = "0")]
+pub fn run_window_configuration_change_hook(frame: Option<LispObject>) -> LispObject {
+    let f = frame.map_or(Qnil, |obj| obj);
+    unsafe { run_window_conf_change_hook(decode_live_frame(f)) };
+    Qnil
+}
+
+#[rustfmt::skip]
```

Comment:
Why the skip?

Reply:
rustfmt causes an error when I try make.

```
$ make
:
make[1]: Entering directory '/path/to/remacs/src'
  GEN      globals.h
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:378:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
/bin/sh: line 1: 1785984 Aborted                 (core dumped) ../lib-src/make-docfile -d . -g dispnew.o frame.o scroll.o xdisp.o menu.o xmenu.o window.o charset.o coding.o category.o ccl.o character.o chartab.o bidi.o cm.o term.o terminal.o xfaces.o xterm.o xfns.o xselect.o xrdb.o xsmfns.o xsettings.o gtkutil.o emacsgtkfixed.o dbusbind.o emacs.o keyboard.o macros.o keymap.o sysdep.o buffer.o filelock.o insdel.o minibuf.o fileio.o dired.o casefiddle.o indent.o search.o regex.o undo.o alloc.o data.o doc.o editfns.o callint.o eval.o fns.o font.o print.o lread.o syntax.o unexelf.o bytecode.o process.o gnutls.o callproc.o region-cache.o sound.o atimer.o doprnt.o intervals.o textprop.o composite.o xml.o lcms.o inotify.o profiler.o thread.o systhread.o sheap.o xfont.o ftfont.o xftfont.o ftxfont.o fontset.o fringe.o image.o xgselect.o json.o ../rust_src/src/*.rs > globals.tmp
make[1]: *** No rule to make target 'gl-stamp', needed by 'globals.h'.  Stop.
make[1]: Leaving directory '/path/to/remacs/src'
make: *** [Makefile:429: src] Error 2
```

And I followed these codes.
https://github.com/remacs/remacs/blob/master/rust_src/src/process.rs#L558-L559

rustfmt seems to broke macro expansion.
Do you have any idea?

This code is formatted to following

```rust
def_lisp_sym!(
    Qwindow_configuration_change_hook,
    "window-configuration-change-hook"
);
```

Reply:
Can you show the backtrace you get when you reproduce the error with `RUST_BACKTRACE=1` set?

Reply:
Here's the backtrace.

```
$ RUST_BACKTRACE=1 make
make -C lib all
make[1]: Entering directory '/path/to/remacs/lib'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/path/to/remacs/lib'
make -C lib-src all
make[1]: Entering directory '/path/to/remacs/lib-src'
make[1]: Nothing to be done for 'all'.
make[1]: Leaving directory '/path/to/remacs/lib-src'
make -C src VCSWITNESS='$(srcdir)/../.git/logs/HEAD' all
make[1]: Entering directory '/path/to/remacs/src'
  GEN      globals.h
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', src/libcore/option.rs:378:21
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:200
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:214
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:384
   8: rust_begin_unwind
             at src/libstd/panicking.rs:311
   9: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
  10: core::panicking::panic
             at src/libcore/panicking.rs:49
  11: core::option::Option<T>::unwrap
             at /rustc/bc2e84ca0939b73fcf1768209044432f6a15c2e5/src/libcore/macros.rs:12
  12: scan_rust_file
             at remacs-lib/docfile.rs:168
/bin/sh: line 1: 2857917 Aborted                 (core dumped) ../lib-src/make-docfile -d . -g dispnew.o frame.o scroll.o xdisp.o menu.o xmenu.o window.o charset.o coding.o category.o ccl.o character.o chartab.o bidi.o cm.o term.o terminal.o xfaces.o xterm.o xfns.o xselect.o xrdb.o xsmfns.o xsettings.o gtkutil.o emacsgtkfixed.o dbusbind.o emacs.o keyboard.o macros.o keymap.o sysdep.o buffer.o filelock.o insdel.o minibuf.o fileio.o dired.o casefiddle.o indent.o search.o regex.o undo.o alloc.o data.o doc.o editfns.o callint.o eval.o fns.o font.o print.o lread.o syntax.o unexelf.o bytecode.o process.o gnutls.o callproc.o region-cache.o sound.o atimer.o doprnt.o intervals.o textprop.o composite.o xml.o lcms.o inotify.o profiler.o thread.o systhread.o sheap.o xfont.o ftfont.o xftfont.o ftxfont.o fontset.o fringe.o image.o xgselect.o json.o ../rust_src/src/*.rs > globals.tmp
make[1]: *** No rule to make target 'gl-stamp', needed by 'globals.h'.  Stop.
make[1]: Leaving directory '/path/to/remacs/src'
make: *** [Makefile:429: src] Error 2
```

Reply:
Ah, I see. When we parse the `def_lisp_sym!` macro in docfile.rs, we can only correctly parse it if it's all on one line. Same with `defvar_lisp!` and the others. For now I suggest leaving the `#[rustfmt::skip]` and we'll deal with docfile.rs separately.

Reply:
@db48x  Thank you!
You meant, this regexp doesn't support line breaks?
https://github.com/remacs/remacs/blob/master/rust_src/remacs-lib/docfile.rs#L166

```diff
-def_lisp_sym!\((.+?),\s+"(.+?)"\);
+def_lisp_sym!\(\s*(.+?),\s*"(.+?)"\s*\);
```

I found that `defvar_lisp!` has the same issue.